### DISCUSSION TITLE
web-authn with localsend?

### DISCUSSION THREAD

**Discussion by ghost** [localsend/localsend#102] [Discussion (English, International)] [answered]
Hi everyone.

Would it be possible to authenticate devices through the browser with web-authn along with the local send protocol locally?

*What do you all think of the idea of adding web-authn to localsend?*

*what is web-authn?*
WebAuthn is the API standard that allows servers, applications, websites, and other systems to manage and verify registered users with passwordless authentication such as a biometric or possession-based device authenticator.

**Comment by Tienisto** [comment, marked as answer]
Currently, there is no authentication process, only "Accept" or "Decline" so I don't know. How should the workflow look like in your opinion?

**Reply by ghost** [reply]
Hi Tienisto. 

> Currently, there is no authentication process, only "Accept" or "Decline" so I don't know. 

thank you for feedback here.

> How should the workflow look like in your opinion?

- I would like to verify the device through biometrics and/or a password generated to click on the device. 
- Things like Bluetooth have a feature of verifying the connection to the device through an autogenerated and temporary password. 
- In my opinion, this is a good workflow for localsend: biometrics and/or autogenerated and temporary password for verifying the connection to the device.

*what do you think of this idea: biometrics and/or autogenerated and temporary password for verifying the connection to the device?*

<img src="https://user-images.githubusercontent.com/111701513/216737717-be86ebb9-ed36-4782-8f2e-597d841f3f9b.png" alt="image" width="300px"/>

**Reply by ghost** [reply]
Hi.

There is a library called: [react-native-touch-id](https://www.npmjs.com/package/react-native-touch-id) - that's exactly it, the touch id serves to validate things. This image above is the touch-id. For example,

![](https://cloud.githubusercontent.com/assets/1627824/7975919/2c69a776-0a42-11e5-9773-3ea1c7dd79f3.gif)

With the touch-id it would be possible to validate the file at each connection or when a connection is initiated either with something biometric or password.
